arduino/test/health_monitor_core_test.cpp

```diff
@@ -0,0 +1,263 @@
+    int treament_events_handled = 0;
+    game_update(
+        player,
+        [&iterations]() -> Event {
+            const auto it = iterations++;
+            if (it < 1) {
+                return Event{ExposureEvent{.human = 3, .cat = 1}};
+            } else {
+                return Event{};
+            }
+        },
+        [&exposure_events_handled](ExposureEvent const& exposure_event) {
+            ++exposure_events_handled;
+        },
+        [&treament_events_handled](TreatmentEvent const& exposure_event) {
+            ++treament_events_handled;
+        });
+
+    ASSERT_EQ(exposure_events_handled, 1);
+    ASSERT_EQ(treament_events_handled, 0);
+
+    ASSERT_GT(player.tick, player_init.tick);
+    EXPECT_EQ(player.tick, 1);
+
+    ASSERT_GT(player.health.health, player_init.health.health);
+    EXPECT_GT(player.health.health, to_health(StateBounds::HEALTHY));
+}
+
+TEST(HealthMonitorCoreTests, SingleExposureEventThenSingleTreatmentEvent) {
+    const PlayerState player_init = new_player_state();
+    PlayerState player            = player_init;
+
+    int iterations              = 0;
+    int exposure_events_handled = 0;
+    int treament_events_handled = 0;
+    game_update(
+        player,
+        [&iterations]() -> Event {
+            const auto it = iterations++;
+            if (it < 1) {
+                return Event{ExposureEvent{.human = 3, .cat = 1}};
+            } else if (it < 2) {
+                return Event{TreatmentEvent{}};
+            } else {
+                return Event{};
+            }
+        },
+        [&exposure_events_handled](ExposureEvent const& exposure_event) {
+            ++exposure_events_handled;
+        },
+        [&treament_events_handled](TreatmentEvent const& exposure_event) {
+            ++treament_events_handled;
+        });
+
+    ASSERT_EQ(exposure_events_handled, 1);
+    ASSERT_EQ(treament_events_handled, 1);
+
+    ASSERT_GT(player.tick, player_init.tick);
+    EXPECT_EQ(player.tick, 1);
+
+    ASSERT_EQ(player.health.health, player_init.health.health);
+    EXPECT_EQ(player.health.health, to_health(StateBounds::SUPER_HEALTHY));
+}
+
+TEST(HealthMonitorCoreTests, ManyExposureEventsToZombie) {
+    const PlayerState player_init = new_player_state();
+    PlayerState player            = player_init;
+
+    int exposure_events_handled = 0;
+    int treament_events_handled = 0;
+    game_update(
+        player,
+        [&player]() -> Event {
+            if (!is_zombie(player.health.health)) {
+                return Event{ExposureEvent{.human = 3, .cat = 1}};
+            } else {
+                return Event{};
+            }
+        },
+        [&exposure_events_handled](ExposureEvent const& exposure_event) {
+            ++exposure_events_handled;
+        },
+        [&treament_events_handled](TreatmentEvent const& exposure_event) {
+            ++treament_events_handled;
+        });
+
+    ASSERT_EQ(exposure_events_handled, 49);
```

Comment:
Doesn't this seem to conflict with https://github.com/kestrel-labs/superspreader/pull/29/files#diff-5c8a33aae609be46d8303b3160f1cf2271548fc48c3d8523487231594928b9e6R109

Like I would expect that test to take less iterations than this one.

And of course, the comment about this test being misleading